### PR TITLE
I want to use the Browse Everything gem

### PR DESCRIPTION
**Pull request by banurekha**
Pull request for browse everything gem implementation. This covers all sub task that are closed within the store https://jira.duraspace.org/browse/HYDRASIR-218

### CODE REVIEW

File: app/services/curation_concern/base_actor.rb [hunk 1]
```diff
@@ -4,12 +4,13 @@ module CurationConcern
...
   # * #delete
   class BaseActor
-    attr_reader :curation_concern, :user, :attributes
+    attr_reader :curation_concern, :user, :attributes, :cloud_resources
     def initialize(curation_concern, user, input_attributes)
       @curation_concern = curation_concern
       @user = user
       @attributes = input_attributes.dup.with_indifferent_access
       @visibility = attributes[:visibility]
+      @cloud_resources= attributes.delete(:cloud_resources.to_s)
```
Comment (jeremyf) [inline app/services/curation_concern/base_actor.rb:13]: Why `:cloud_resources.to_s`; Why not just `attributes.delete('cloud_resources')`

File: app/controllers/curation_concern/generic_files_controller.rb [hunk 2]
```diff
@@ -17,6 +18,20 @@ def action_name_for_authorization
   end
   protected :action_name_for_authorization
 
+  def cloud_resources_valid?
+    if params.has_key?(:selected_files) && params[:selected_files].length>1
```
Comment (jeremyf) [inline app/controllers/curation_concern/generic_files_controller.rb:22]: `params[:selected_files].present?` will do the same thing as the above two checks.

File: config/browse_everything_providers.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (jeremyf) [inline config/browse_everything_providers.yml]: Are these example provider keys? Because they shouldn't be in the open.

File: app/views/curation_concern/images/_form_files_and_links.html.erb [hunk 4]
```diff
@@ -7,6 +7,7 @@
           </legend>
           <%= f.input :files, as: :file, label: 'Upload your file' %>
         </fieldset>
+        <%#render "form_browse_everything", curation_concern: curation_concern, f: f %>
```
Comment (jeremyf) [inline app/views/curation_concern/images/_form_files_and_links.html.erb]: This is commented out. Does it need to be here?

Comment (banurekha) [inline app/views/curation_concern/images/_form_files_and_links.html.erb]: No. Has to be removed from code. I will delete it on my next pull request.

File: config/browse_everything_providers.yml (hunk 3 above)
Comment (banurekha) [inline config/browse_everything_providers.yml]: I think decision was made to remove this file. I will remove it

File: app/controllers/curation_concern/generic_files_controller.rb (hunk 2 above)
Comment (banurekha) [inline app/controllers/curation_concern/generic_files_controller.rb:22]: Need to error out only if more than one file is selected from cloud resource. I dont think present? will do that

**Comment by nomadicoder** [comment]
Attempted to upload file with Browse Everything configured for local file system. I selected a file and started upload. Javascript encountered an error: "Uncaught TypeError: Cannot read property 'opts' of null." File was not added.

**Comment by banurekha** [comment]
Can you please give me more details? any screenshot.. are you testing on the branch feature/HYDRASIR-281?

File: app/controllers/curation_concern/generic_files_controller.rb (hunk 2 above)
Comment (jeremyf) [inline app/controllers/curation_concern/generic_files_controller.rb:22]: Sorry. I misread `>1` as `>0`. You are right.

**Comment by nomadicoder** [comment]
I performed the following in Safari:

```
checkout feature/HYDRASIR-218
bundle install
bundle exec rake regenerate
cd spec/internal
rake jetty:clean
rake jetty:start
bundle exec rails s
```
- Visited `0.0.0.0:3000`
- Created an account, agreed to terms, and updated profile
- Visited `0.0.0.0:3000/remote_files/browse`
- Clicked on `File System`
- Clicked on `features.rb`
- Clicked on "Submit"
- "Submit" button changed to "Loading" and pointer changed to perpetual spinner.

When I tried to submit a file with Chrome, the Javascript console showed the following error:

```
Uncaught TypeError: Cannot read property 'opts' of null behavior.js?body=1:86
  (anonymous function) behavior.js?body=1:86
  fire jquery.js?body=1:3100
  self.fireWith jquery.js?body=1:3212
  done jquery.js?body=1:9311
  callback jquery.js?body=1:9721
```

I also created a local `config/browse_everything_providers.yml` file that would point to an asset directory on my development system. I attempted the same steps with a `.jpg` file and encountered the same problem.

screen shot 2014-03-27 at 9 11 24 am

**Comment by banurekha** [comment]
browse_everything routes are never meant to be accessed directly. It always sit as modal window on top of form. And hence the error.

So here are the steps to use/test this pull request
- Visited 0.0.0.0:3000
- Created an account, agreed to terms, and updated profile
- Visited 0.0.0.0:3000/concern/generic_works/new (although I keep getting Unregistered predicate: :has_editor: May be side effect of different commit. Not sure)
- select big green button called 'Browse!'
- Select the file from file system or any provider you like
- Click submit
- See a message on number of files selected
- Proceed with filling the form for work
- Agree to terms and submit the form

This should ingested the selected file to fedora.

**Comment by rbalekai** [comment]
Unregistered predicate: :has_editor: May be side effect of different commit.

@banurekha, you have to regenerate the spec/internals. I added predicate_mappings.yml file to it. The Group and Editors will expect that to be in place.

**Comment by nomadicoder** [comment]
I followed the revised directions and got a server error
screen shot 2014-03-27 at 9 46 20 am

**Comment by banurekha** [comment]
Found that error later today and pushed the changes to the branch earlier.. Just do a pull.. Should have informed..

**Comment by banurekha** [comment]
@rbalekai haven't do that yet.. will let you later once I do that. Thanks for letting me know that

**Comment by nomadicoder** [comment]
@banurekha, Thank you. Local files works now. Just have a problem with the browse button where I have to reload the page to make it work. I also need to get my Cloud storage api keys straightened out so I can try the cloud services.

**Comment by banurekha** [comment]
This pull request was closed without merging... Should I resend it?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -152,7 +152,7 @@ However, a well formed Gherkin-syntax story is a suitable proxy; Especially if y
 
 We encourage the use of [custom rspec matchers](https://www.relishapp.com/rspec/rspec-expectations/v/2-3/docs/custom-matchers/define-matcher) as this helps our specs better convey what is going on.
 
-To find the existing Rspec matchers in this project, you can run `rake spec:show_matchers`.
+To find the existing Rspec matchers in this project, you can run `rake yenta`.
```
Comment (jeremyf) [inline CONTRIBUTING.md:155]: I'm not sure how this made it into the commits.

File: app/controllers/catalog_controller.rb [hunk 6]
```diff
@@ -68,9 +78,8 @@ def self.search_config
 
     # solr fields that will be treated as facets by the blacklight application
     #   The ordering of the field names is the order of the display
-    config.add_facet_field solr_name("human_readable_type", :facetable), label: "Type of Work", limit: 5
+    config.add_facet_field solr_name("human_readable_type", :facetable), label: "Type of Work", :query => human_readable_type_hash, limit: 5
```
Comment (jeremyf) [inline app/controllers/catalog_controller.rb:81]: Why is this here?
